### PR TITLE
Inject APP_PROTOCOL in app containers

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Changes the Dapr Injector to inject the APP_PROTOCOL variable into the app containers. This will contain the value of the (computed) `--app-protocol` flag, such as "http", "grpc", "https", "grpcs", or "h2c".

This can be used by SDKs (and users) to determine if they need to start a server with TLS or with H2C support, for example. It may be possible for the SDKs to use this to determine if they should use HTTP or gRPC too, possibly.

See: https://github.com/dapr/dapr/issues/6035#issuecomment-1590624276

### CODE REVIEW

File: pkg/injector/sidecar/consts.go [hunk 1]
```diff
@@ -31,9 +31,16 @@ const (
 	SidecarMetricsEnabledLabel     = "dapr.io/metrics-enabled"
 	APIVersionV1                   = "v1.0"
 	UnixDomainSocketVolume         = "dapr-unix-domain-socket" // Name of the Unix domain socket volume.
+	UserContainerAppProtocolName   = "APP_PROTOCOL"            // Name of the variable exposed to the app containing the app protocol.
```
Comment (JoshVanL) [inline pkg/injector/sidecar/consts.go:34]:
```suggestion
	UserContainerAppProtocolName   = "DAPR_APP_PROTOCOL"            // Name of the variable exposed to the app containing the app protocol.
```

**Review by JoshVanL — CHANGES_REQUESTED** [review]

File: pkg/injector/sidecar/consts.go (hunk 1 above)
Comment (ItalyPaleAle) [reply pkg/injector/sidecar/consts.go:34]: We use `DAPR_*` for env vars that are about the Dapr container, such as "DAPR_HTTP_PORT" -> the port Dapr's HTTP server is listening on. This instead is about the app. We already have "APP_CHANNEL_ADDRESS", "APP_ID", "APP_PORT".

Comment (JoshVanL) [reply pkg/injector/sidecar/consts.go:34]: All of these are still Dapr related, and therefore must have a `DAPR` prefix. Makes sense to have the `DAPR_APP` namespace for apps using Dapr. We should deprecate these other `APP_*` env vars, in favor of `DAPR_APP_*` imo.

In my opinion, not namespacing env vars is quite hostile/bad mannered to users and generally breaks convention, especially with `APP` being so generic.

Comment (JoshVanL) [reply pkg/injector/sidecar/consts.go:34]: Sounds good to me :+1:

**Review by yaron2 — APPROVED** [review]

**Comment by ItalyPaleAle** [comment]
@yaron2 why did you change the title to `DAPR_APP_PROTOCOL`? The env var is `APP_PROTOCOL` because `DAPR_*` things are read by the sidecar (like `APP_PORT`, `APP_API_TOKEN`, etc)

**Comment by yaron2** [comment]
> @yaron2 why did you change the title to `DAPR_APP_PROTOCOL`? The env var is `APP_PROTOCOL` because `DAPR_*` things are read by the sidecar (like `APP_PORT`, `APP_API_TOKEN`, etc)

Misread your discussion with @JoshVanL

**Pull request merged**